README.md

```diff
@@ -1,60 +1,72 @@
+`eslines` helps you to post-process an ESLint JSON report.
+
+The default behavior is transforming all errors into warnings, except 1) those reported in lines modified within the current git branch and 2) any `no-unused-vars` ESLint rule break.
 
 ## How to use it
 
 Install it:
 
 	npm install eslines
 
-Add the default `.eslines.json` config file in you git repo:
-
-	{
-		"rulesNotToDowngrade": [],
-		"remote": "origin/master",
-		"processors": {
-			"default": "lines-modified"
-		}
-	}
+Add the default `.eslines.json` config file to your git repo:
+
+    {
+        "branches": {
+	        "default": ["lines-modified", "enforce"]
+        },
+        "processors": {
+            "lines-modified": {
+                "remote": "origin/master",
+            },
+            "enforce": {
+                "rules": ["no-unused-vars"],
+            }
+        }
+    }
 
 Run it:
 
 	eslint -f json . | eslines
 
 ## Config file
 
-`eslines` reads its configuration from a file named `.eslines.json` placed in the root of your git repository.
-
-`eslines` has two ways of processing an ESLint report, we call them processors. The processor `lines-modified` will downgrade ESLint `errors` to `warnings`, except for lines modified in the current branch. The processor `parsing-errors` takes the ESLint report and outputs a new one containing only `Parsing Errors`.
+`eslines` reads its configuration from a file named `.eslines.json` placed in the root of your git repository. Out of the box, it comes with three ways of post-processing an ESLint report - we call them *processors*.
 
-Options:
-
-* **rulesNotToDowngrade**: an array containing valid [ESLint rule ids](http://eslint.org/docs/rules/). For any rule declared in this array, the `lines-modified` processor won't downgrade to `warnings` reported `errors` for these rules, no matter whether the line was modified or not.
-
-* **remote**: lines modified are determined by diffing this remote git branch against the current branch.
-
-* **processors**: `eslines` will choose the processor to use by default and which one to use for particular branches looking at this object.
+The processor `lines-modified` will transform ESLint `errors` into `warnings`, except for errors that are in lines modified in the current branch. The processor `parsing-errors` takes the ESLint report and outputs a new one containing only `Parsing Errors`. The processor `enforce` will transform any `warning` into an `error`. Processors are composable.
 
 For example:
 
 	{
-		"rulesNotToDowngrade": [],
-		"remote": "origin/master",
+		"branches": {
+			"default": ["lines-modified", "enforce"],
+			"master": ["parsing-errors"],
+			"my/topic-branch": ["parsing-errors"],
+		},
 		"processors": {
-			"default": "lines-modified",
-			"master": "parsing-errors",
-			"my/topic-branch": "parsing-errors"
+			"lines-modified": {
+				"remote": "origin/master",
+			},
+			"enforce": {
+				"rules": ["no-unused-vars"],
```

Comment:
I'm not very fond of the idea of making `enforce` a separate processor independent from `lines-modified`. Downgrade some errors to warnings in one processor and the upgrade them back in another one?

Having it as a config option makes more sense: the `no-unused-vars` rule is special because a change on some line (removing last usage of a variable) triggers a linting error on _another_ line (where the variable is declared). Except for this subtle interplay with `lines-modified`, the `no-unused-vars` is not special at all.